From 761a6928b018d9f1717dafe75c1af1e881b565ea Mon Sep 17 00:00:00 2001
From: Bernd Schmidt <bernds_cb1@t-online.de>
Date: Sun, 20 Oct 2019 14:17:23 +0200
Subject: [PATCH] Disable game tree header with Qt < 5.10

It seems Qt 5.9 is still fairly widespread (issue #22), so it's better to just
disable a non-critical feature rather than require a higher version.

I tried reverting 7c58b67d, which added the current header view and replaced
the one based on QHeaderView, but it now doesn't work for me even under Linux.
The headers are too wide, an issue I previously was seeing only on Windows.
---
 src/gametree.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/gametree.cpp b/src/gametree.cpp
index d6c1bd7..9a5a82a 100644
--- a/src/gametree.cpp
+++ b/src/gametree.cpp
@@ -125,8 +125,12 @@ void GameTree::set_board_win (MainWindow *win, QGraphicsView *header)
 	QScrollBar *hscr = horizontalScrollBar ();
 	QScrollBar *hscr2 = m_header_view->horizontalScrollBar ();
 	connect (hscr, &QScrollBar::valueChanged, hscr2, &QAbstractSlider::setSliderPosition);
+#if (QT_VERSION >= QT_VERSION_CHECK(5, 10, 0))
 	connect (verticalScrollBar (), &QAbstractSlider::rangeChanged,
 		 [this] (int, int) { QMetaObject::invokeMethod (this, &GameTree::resize_header, Qt::QueuedConnection); });
+#else
+	header->hide ();
+#endif
 }
 
 void GameTree::update_prefs ()
-- 
2.23.0

